### PR TITLE
feat: gcp support for storage

### PR DESCRIPTION
**Pull request by naman-modi**
## What does this PR do?
- Added GCP support for storage.

## How does this PR work? (optional)
- Added an additional `StorageType` for gcp.
- Created a new module in storage manager, `GCPStorageUtils` to enable bucket creation, etc.
- Unified the bucket_name in config to a singular `cloud_storage_bucket_name`, rather than having separate for each cloud provider

## Refer to a related PR or issue link
- N.A.

## Checklist

- [X]  I have written the necessary rustdoc comments.
- [X]  I have added the necessary unit tests and integration tests. (for non-documentation changes)

### CODE REVIEW

File: coredb/src/storage_manager/gcp_storage_utils.rs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+          "Project ID is missing",
+        )));
+      }
+    };
+
+    let client = Client::new(config);
+
+    Ok(Self {
+      client: client,
+      region: "US-EAST1".to_owned(),
```
Comment (naman-modi) [inline coredb/src/storage_manager/gcp_storage_utils.rs:41]: Hey @vinaykakade, do we want to shift to config?

File: coredb/src/storage_manager/storage.rs [hunk 2]
```diff
@@ -48,6 +51,22 @@ impl Storage {
 
       // Local storage.
       StorageType::Local => Arc::new(LocalFileSystem::new()),
+
+      // GCP Cloud Storage, creates a bucket if it does not exists
```
Comment (vinaykakade) [inline coredb/src/storage_manager/storage.rs:55]: nit: exists->exist

File: coredb/src/storage_manager/gcp_storage_utils.rs (hunk 1 above)
Comment (vinaykakade) [reply coredb/src/storage_manager/gcp_storage_utils.rs:41]: Good idea. Please move (for aws too - right?)

**Review by vinaykakade — APPROVED** [review]

File: coredb/src/storage_manager/gcp_storage_utils.rs (hunk 1 above)
Comment (naman-modi) [reply coredb/src/storage_manager/gcp_storage_utils.rs]: Yes Vinay, have refactored some part of storage manager to allow this.

File: coredb/src/storage_manager/storage.rs (hunk 2 above)
Comment (naman-modi) [reply coredb/src/storage_manager/storage.rs]: Fixed it

**Pull request merged**